src/include/rbtree.h

```diff
@@ -1,17 +1,207 @@
+/*
+ * The primary function of red-black trees is to provide a fast (O(ln(n))) and cost-effective method for mapping address ranges to numbers and accessing them by specifying an address within that range.
+ * Each RB-node includes two additional pointers, "start" and "end," to denote a memory range.
+ * When a range of memory is mapped from the system, it is recorded by multiple red-black trees with the information of that memory range.
+ * 
+ * In fact, there are foru red-black tree in custommem.c: memprot, mapallmem, mmapmem and blockstree. 
+ * The "blockstree" specifically contains memory ranges for blocks, where each node represents a different block and "data" represent the index of the block in blocklist_s array "p_blocks".
+ * The other three trees serve as general-purpose memory tracking mechanisms. 
+ * In "memprot", data represents the permissions of that memory range, and the function rb_set can set or change permissions for a range of memory.
+ * In the case of "mapallmem" and "mmapmem", "data" can be '1' or '2', depending on factors that are not specified.
```

Comment:
This is false for now. Also, `mapallmem` contains `mmapmem` (i.e. `mapallmem` is 1 if `mmapmem` is 1), so if/when `mapallmem` and `mmapmem` are merged, `data` could take values `0` (nowhere), `1` (only in `mapallmem`) or `2` (`mapallmem` + `mmapmem`).
Note that there is still the question of whether this merge should happen.